style(components): change button state styling to fix disappearing buttons

shanegrouber opened the pull request:
- Update button classes to include hover and enabled states
- Remove unused CSS rules to streamline styles

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Enhanced button styling across various components for improved visual feedback during user interactions.

- **Bug Fixes**
	- Simplified rendering of the Chatbot component by removing unnecessary wrapper, potentially improving performance and layout.

- **Documentation**
	- Updated CSS classes for better theming and responsiveness in dark mode.

- **Chores**
	- Removed unused components and import statements to streamline the codebase.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

shanegrouber committed:
style(components): enhance button state styling for better UX

- Update button classes to include hover and enabled states
- Remove unused CSS rules to streamline styles

(your buttons now have more states than a soap opera character)

Omri-Levy commented on apps/backoffice-v2/src/domains/notes/Notes.tsx:
```diff
@@ -88,7 +88,7 @@ export const Notes = ({
                 aria-disabled={isLoading}
                 className={
-                  'mt-3 h-5 self-end p-4 text-sm font-medium aria-disabled:pointer-events-none aria-disabled:opacity-50'
+                  'mt-3 h-5 self-end p-4 text-sm font-medium enabled:bg-slate-800 enabled:hover:bg-slate-800/90 aria-disabled:pointer-events-none aria-disabled:opacity-50'
```
Let's use bg-primary over slate directly. That's the default color for Button.

Omri-Levy commented on apps/backoffice-v2/src/lib/blocks/components/CallToActionLegacy/CallToActionLegacy.tsx:
```diff
@@ -189,7 +189,10 @@ export const CallToActionLegacy: FunctionComponent<ICallToActionLegacyProps> = (
-            className={ctw({ 'flex gap-2': isReuploadResetable })}
+            className={ctw({
+              'flex gap-2': isReuploadResetable,
+              'enabled:bg-warning enabled:hover:bg-warning/90': !isReuploadResetable,
```
Can use 'enabled' directly without JS. ctw accepts multiple arguments and it can also receive arrays. If we wanted to keep this line the same passing `true` would be less misleading as this class does not apply when the button is disabled regardless of the condition. https://www.npmjs.com/package/clsx

Omri-Levy commented on apps/backoffice-v2/src/pages/Entity/components/Case/actions-variants/DefaultActions/DefaultActions.tsx:
```diff
@@ -35,7 +35,11 @@ export const DefaultActions = () => {
+            className={ctw(
+              { loading: debouncedIsLoadingRejectCase },
+              'whitespace-nowrap',
+              'enabled:bg-warning enabled:hover:bg-warning/90',
```
Use this as reference for the other places with enabled. First argument is an object and second is string literal. 🙂

shanegrouber committed:
fix(buttons): update background color for button components

- Change button styles from slate to primary background and hover colors
- Ensure consistency across different components for better UX

(your color choices are so dark, they could be the backdrop for a horror movie)

Omri-Levy reviewed (approved):
Great job. 💪🏼

Omri-Levy committed:
Merge branch 'dev' into BAL-3165

shanegrouber committed:
Merge branch 'dev' into BAL-3165
